net.go

```diff
@@ -425,6 +426,122 @@ func (e *EurekaConnection) ScheduleSecureVIPAddressUpdates(addr string, await bo
 	return e.scheduleVIPAddressUpdates(addr, await, done, options), nil
 }
 
+// An InstanceSetSource holds a periodically updated set of instances registered with Eureka.
+type InstanceSetSource struct {
+	m         *sync.RWMutex
+	instances []*Instance
+	done      chan<- struct{}
+}
+
+func newInstanceSetSourceFromChannel(await bool, updates <-chan InstanceSetUpdate, done chan<- struct{}) *InstanceSetSource {
```

Comment:
When I first started writing this, I was focused on Eureka's `com.netflix.discovery.shared.Applications` class, which has separate methods and fields for dealing with insecure and secure VIP addresses. Here are a few examples:
- `Applications#getInstancesByVirtualHostName()`  
  Note how at this layer, they're not called "virtual _IP addresses_", but rather host names. Someone changed their mind as Eureka evolved.
- `Applications#getInstancesBySecureVirtualHostName()`
- `Applications#virtualHostNameAppMap`
- `Applications#secureVirtualHostNameAppMap`
- `Applications#virtualHostNameMap`
- `Applications#secureVirtualHostNameIndexMap`
- `Applications#virtualHostNameMap`
- `Applications#secureVirtualHostNameMap`

I had thought that it was important to treat insecure VIP addresses and secure VIP addresses as two different kinds of things. However, as we move upward in Eureka's interface to `DiscoveryClient`, we find that it agrees with your proposal:
- `DiscoveryClient#getInstancesByVipAddress(String vipAddress, boolean secure)`
- `DiscoveryClient#getInstancesByVipAddress(String vipAddress, boolean secure, String region)`
- `DiscoveryClient#getInstancesByVipAddressAndAppName(String vipAddress, String appName, boolean secure)`

Given that, I'm willing to condense these function pairs back down to the internal common functions to which they defer. Do you think that the insecure v. secure distinction should be conveyed via a raw `bool` parameter, or should I export [the internal `vipAddressKind ` type from file _net.go_](https://github.com/hudl/fargo/pull/57/files#diff-e2893928b1f232947cf45a36fdcb6f9b)? Callers would then write `farge.Insecure` or `fargo.Secure` rather than `false` or `true`.